src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -384,15 +387,25 @@ impl<'a, 'tcx> DocFolder for LinkCollector<'a, 'tcx> {
             None
         };
 
-        // FIXME: get the resolver to work with non-local resolve scopes.
-        let parent_node = self.cx.as_local_hir_id(item.def_id).and_then(|hir_id| {
-            // FIXME: this fails hard for impls in non-module scope, but is necessary for the
-            // current `resolve()` implementation.
-            match self.cx.as_local_hir_id(self.cx.tcx.parent_module(hir_id).to_def_id()).unwrap() {
-                id if id != hir_id => Some(id),
-                _ => None,
+        use rustc_middle::ty::DefIdTree;
+        let parent_node = if item.is_fake() {
+            // FIXME: is this correct?
```

Comment:
I _suspect_ we'll never hit this case, it's only for auto traits and blanket impls, which don't get docs resolved on the local scope anyway.

What happens if you panic here? Especially if you try and document a blanket impl that uses intra doc links?

Reply:
Rustdoc panics if I don't check this (with or without an explicit panic):

```
thread 'rustc' panicked at 'index out of bounds: the len is 38899 but the index is 38905', /home/joshua/src/rust/src/librustc_hir/definitions.rs:53:9
  18: rustc_hir::definitions::DefPathTable::def_key
             at /home/joshua/src/rust/src/librustc_hir/definitions.rs:53
  19: rustc_metadata::rmeta::decoder::<impl rustc_metadata::creader::CrateMetadataRef>::def_key
             at src/librustc_metadata/rmeta/decoder.rs:1411
  20: rustc_metadata::rmeta::decoder::cstore_impl::<impl rustc_middle::middle::cstore::CrateStore for rustc_metadata::creader::CStore>::def_key
             at src/librustc_metadata/rmeta/decoder/cstore_impl.rs:484
  21: rustc_middle::ty::context::TyCtxt::def_key
             at src/librustc_middle/ty/context.rs:1188
  22: <rustc_middle::ty::context::TyCtxt as rustc_middle::ty::DefIdTree>::parent
             at src/librustc_middle/ty/mod.rs:358
  23: <rustdoc::passes::collect_intra_doc_links::LinkCollector as rustdoc::fold::DocFolder>::fold_item
             at src/librustdoc/passes/collect_intra_doc_links.rs:482
```

This was the motivation for #73098.

Reply:
And actually it will panic on _any_ file, including empty files.